gui/elements/_viewer.py

```diff
@@ -147,31 +145,31 @@ def _update_layers(self):
         """
         for layer in self.layers:
             layer._set_view_slice(self.dimensions.indices)
-
-        self._update_statusBar()
+        self.dimensions._update_index(None)
+        self._update_status_bar()
 
     def _on_layers_change(self, event):
         """Called whenever a layer is changed.
         """
         self.dimensions._child_layer_changed = True
         self.dimensions._update()
 
-    def _set_annotation_mode(self, bool):
+    def _set_annotation(self, bool):
         if bool:
             self.annotation = True
             self._qt.view.interactive = False
             if self._active_markers:
-                self._qt.canvas.native.setCursor(Qt.CrossCursor)
+                self._qt.canvas.native.setCursor(self._qt._cursors['cross'])
```

Comment:
Even ignoring my earlier comment, this abstraction would seem more useful if we could get to:

```
self.canvas.native.set_cursor(self._cursors['cross'])
```

ie, remove all mention of Qt, so that if we wanted to rip it out, we could do it inside a canvas abstraction, or something similar.

Reply:
(Or maybe `self._gui.canvas....`)

Reply:
If the `_cursors` were on the `Viewer` class then I'd need to be importing from pyqt inside the `Viewer` class which we don't want. Instead they are being imported inside the `QtViewer` class which is better. I think the problem here is that we are adding the Qt stuff onto the viewer as `_qt`, which if we wanted to rip Qt out and replace it would make the naming strange. Instead we could have a different name for that variable so it would be `self._foo.cursors` and `self._foo` could be a Qt object or another object. (Note we'd need a better name than `foo`).

Reply:
on reflection now it seems like all this should be happening inside the Qt elements and listening to events from the viewer. Will work on fixing once #92 gets resolved

Reply:
I have fixed this